### PR TITLE
dex: 0.7 -> 0.8.0

### PR DESCRIPTION
**Pull request by kyrias**
###### Motivation for this change
Upstream update

###### Things done

- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)

---

### CODE REVIEW

File: pkgs/tools/X11/dex/default.nix [hunk 1]
```diff
@@ -1,18 +1,19 @@
-{ stdenv, fetchFromGitHub, python3 }:
+{ stdenv, fetchFromGitHub, python3, sphinx }:
```
Comment (xeji) [inline pkgs/tools/X11/dex/default.nix]: no need to add `sphinx` here, you can reference it as `python3.pkgs.sphinx`, so the argument `python3` is enough.

File: pkgs/tools/X11/dex/default.nix [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
   src = fetchFromGitHub {
     owner = "jceb";
     repo = program;
     rev = "v${version}";
-    sha256 = "041ms01snalapapaniabr92d8iim1qrxian626nharjmp2rd69v5";
+    sha256 = "13dkjd1373mbvskrdrp0865llr3zvdr90sc6a6jqswh3crmgmz4k";
   };
 
   propagatedBuildInputs = [ python3 ];
+  nativeBuildInputs = [ sphinx ];
```
Comment (xeji) [inline pkgs/tools/X11/dex/default.nix]: `python3.pkgs.sphinx`

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -2063,7 +2063,9 @@ with pkgs;
     gtk = gtk3;
   };
 
-  dex = callPackage ../tools/X11/dex { };
+  dex = callPackage ../tools/X11/dex {
+    sphinx = python3Packages.sphinx;
```
Comment (xeji) [inline pkgs/top-level/all-packages.nix]: This is not necessary, see comments above.

File: pkgs/tools/X11/dex/default.nix (hunk 1 above)
Comment (kyrias) [reply pkgs/tools/X11/dex/default.nix]: Ah, I just copied it because I saw some other packages doing it that way, but fixed now.

**Comment by xeji** [comment]
@GrahamcOfBorg build dex

**Comment by xeji** [comment]
Thank you!

**Pull request merged**